### PR TITLE
kubeadm: add Priority to admission control

### LINKED ISSUE #434: Enable the Priority admission plugin in v1.9
ref: https://github.com/kubernetes/kubernetes/pull/49322#issuecomment-328625401

### PR DESCRIPTION
**Pull request by andrewsykim**
**What this PR does / why we need it**:
Adds Priority admission control to kubeadm for all kubernetes versions > v1.9 alpha.

Related: https://github.com/kubernetes/kubernetes/pull/49322

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*:
fixes https://github.com/kubernetes/kubeadm/issues/434

**Special notes for your reviewer**:

**Release note**:
```release-note
Enable Priority admission control in kubeadm. 
```

cc @luxas

### CODE REVIEW

File: cmd/kubeadm/app/constants/constants.go [hunk 1]
```diff
@@ -214,6 +214,9 @@ var (
 
 	// MinimumKubeletVersion specifies the minimum version of kubelet which kubeadm supports
 	MinimumKubeletVersion = version.MustParseSemantic("v1.8.0-beta.0")
+
+	// MinimumUnstableVersion specifies the earliest version of the upcoming major release
+	MinimumUnstableVersion = version.MustParseSemantic("v1.9.0-alpha.1")
```
Comment (andrewsykim) [inline cmd/kubeadm/app/constants/constants.go]: Changed the version for other constants in https://github.com/kubernetes/kubernetes/pull/53181

File: cmd/kubeadm/app/phases/controlplane/manifests.go [hunk 2]
```diff
@@ -40,7 +40,11 @@ import (
 const (
 	DefaultCloudConfigPath = "/etc/kubernetes/cloud-config"
 
-	defaultv17AdmissionControl = "Initializers,NamespaceLifecycle,LimitRanger,ServiceAccount,PersistentVolumeLabel,DefaultStorageClass,DefaultTolerationSeconds,NodeRestriction,ResourceQuota"
+	// admission controllers for any version <= current stable release
+	// for kubeadm this is the current stable release and and 1 major release behind
+	defaultStableAdmissionControl = "Initializers,NamespaceLifecycle,LimitRanger,ServiceAccount,PersistentVolumeLabel,DefaultStorageClass,DefaultTolerationSeconds,NodeRestriction,ResourceQuota"
+	// admission controllers > current stable release
+	defaultUnstableAdmissionControl = "Initializers,NamespaceLifecycle,LimitRanger,ServiceAccount,PersistentVolumeLabel,DefaultStorageClass,DefaultTolerationSeconds,NodeRestriction,ResourceQuota,Priority"
```
Comment (luxas) [inline cmd/kubeadm/app/phases/controlplane/manifests.go]: I'm not sure where in the chain this should be, but I'm pretty sure it should be early in the chain. Please search for how it's set up in GCE/GKE

File: cmd/kubeadm/app/phases/controlplane/manifests.go [hunk 3]
```diff
@@ -40,7 +40,11 @@ import (
 const (
 	DefaultCloudConfigPath = "/etc/kubernetes/cloud-config"
 
-	defaultv17AdmissionControl = "Initializers,NamespaceLifecycle,LimitRanger,ServiceAccount,PersistentVolumeLabel,DefaultStorageClass,DefaultTolerationSeconds,NodeRestriction,ResourceQuota"
+	// admission controllers for any version <= current stable release
+	// for kubeadm this is the current stable release and and 1 major release behind
+	defaultStableAdmissionControl = "Initializers,NamespaceLifecycle,LimitRanger,ServiceAccount,PersistentVolumeLabel,DefaultStorageClass,DefaultTolerationSeconds,NodeRestriction,ResourceQuota"
```
Comment (luxas) [inline cmd/kubeadm/app/phases/controlplane/manifests.go]: Could we branch this as v18 and v19?
I think that would be a bit better, as the v19 chain will eventually be stable ;)

File: cmd/kubeadm/app/constants/constants.go [hunk 4]
```diff
@@ -218,6 +218,9 @@ var (
 
 	// MinimumKubeletVersion specifies the minimum version of kubelet which kubeadm supports
 	MinimumKubeletVersion = version.MustParseSemantic("v1.8.0")
+
+	// MinimumUnstableVersion specifies the earliest version of the upcoming major release
+	MinimumUnstableVersion = version.MustParseSemantic("v1.9.0-alpha.1")
```
Comment (luxas) [inline cmd/kubeadm/app/constants/constants.go]: MinimumPriorityAdmissionVersion would be a bit more descriptive IMO

**Review by luxas — COMMENTED** [review]
A couple of naming nits and one comment on ordering, please fix.

Other than that, this LGTM

File: cmd/kubeadm/app/phases/controlplane/manifests.go (hunk 2 above)
Comment (andrewsykim) [reply cmd/kubeadm/app/phases/controlplane/manifests.go]: Good catch, will use this as a referene https://github.com/kubernetes/kubernetes/pull/49322/files#diff-57380bb03c3598646d5a3cb66518fa0bL237

File: cmd/kubeadm/app/constants/constants.go (hunk 4 above)
Comment (andrewsykim) [reply cmd/kubeadm/app/constants/constants.go]: If we have v18 and v19 constants for admission controller list than I don't think this is needed.

Comment (luxas) [reply cmd/kubeadm/app/constants/constants.go]: Well, yeah, either way.

**Review by luxas — APPROVED** [review]
/lgtm

Thanks!

**Pull request merged**